File: C++/Programming Basics with C++/09. Exercise - While-loop/simple-tasks/Cake.cpp
#include<iostream>
#include <string>

using namespace std;

int main() {
    int cakeWidth, cakeLength;
    cin >> cakeWidth >> cakeLength;
    int numberOfCakePiece = cakeWidth * cakeLength;
    int inputPieceOfCake = 0;
    string input;
    while (numberOfCakePiece >= inputPieceOfCake && input != "STOP") {
        cin >> input;
        if (input != "STOP") {
            inputPieceOfCake = inputPieceOfCake + stoi(input);
        }
    }
    if (input == "STOP" && numberOfCakePiece >= inputPieceOfCake) {
        cout << numberOfCakePiece - inputPieceOfCake << " pieces are left." << endl;
    } else {
        cout << "No more cake left! You need " << inputPieceOfCake - numberOfCakePiece << " pieces more." << endl;
    }

    return 0;
}

File: C#/Programming Basics with C#/09. While-Loop - Exercise/06. Cake/Cake.cs
using System;

class Cake
{
    static void Main()
    {
        int cakeWidth = int.Parse(Console.ReadLine());
        int cakeLenght = int.Parse(Console.ReadLine());
        int numberOfCakePiece = cakeWidth * cakeLenght;
        int inputPieceOfCake = 0;
        string input = "variable";
        while (numberOfCakePiece >= inputPieceOfCake && input != "STOP")
        {
            input = Console.ReadLine();
            if (input != "STOP")
            {
                inputPieceOfCake = inputPieceOfCake + int.Parse(input);
            }
        }
        if (input == "STOP" && numberOfCakePiece >= inputPieceOfCake)
        {
            Console.WriteLine($"{numberOfCakePiece - inputPieceOfCake} pieces are left.");
        }
        else
        {
            Console.WriteLine($"No more cake left! You need {inputPieceOfCake - numberOfCakePiece} pieces more.");
        }
    }
}
